pkg/api/admin.go

```diff
@@ -19,3 +29,52 @@ func AdminGetStats(c *models.ReqContext) response.Response {
 
 	return response.JSON(200, statsQuery.Result)
 }
+
+func (hs *HTTPServer) getAuthorizedSettings(ctx context.Context, user *models.SignedInUser, bag setting.SettingsBag) (setting.SettingsBag, error) {
+	if hs.AccessControl.IsDisabled() {
+		return bag, nil
+	}
+
+	eval := func(scopes ...string) (bool, error) {
+		return hs.AccessControl.Evaluate(ctx, user, accesscontrol.ActionSettingsRead, scopes...)
+	}
+
+	ok, err := eval(accesscontrol.ScopeSettingsAll)
```

Comment:
Can you add an explanatory comment? Something like:
```suggestion
	// Early return if user has access to all sections
	ok, err := eval(accesscontrol.ScopeSettingsAll)
```